src/westpa/core/binning/assign.py

```diff
@@ -409,6 +409,9 @@ def add_mapper(self, mapper, replaces_bin_at):
 
         # we have updated our list of recursed bins, so set our own start index to trigger a recursive
         # reassignment of mappers' output values
+        # Note that we're reordering the recursion targets based on outer bin numbers (dict keys) first,
+        # so the order the mappers were added no longer matters...
+        self._recursion_targets = {k: self._recursion_targets[k] for k in sorted(self._recursion_targets)}
         self.start_index = self.start_index
```

Comment:
Just out of curiosity, I wonder why this line is even needed?

Reply:
That's the fix (and [seemingly the fastest way to reorder a dictionary](https://stackoverflow.com/a/50569360/17047865)). Otherwise `_recursion_targets` would be in insertion order and start_index would label things differently depending on add order.

Reply:
Whoops, should have been more clear, this line: 
self.start_index = self.start_index

Reply:
Probably because start_index is a property, and how `add_mapper` method can be run recursively, so this ensures that setter is run at every level with have a unique bin indices at creation.

Reply:
ah i see, that makes sense, I didn't realize it was a property